container_registry/container_analysis/src/sample/samples.go

```diff
@@ -0,0 +1,247 @@
+// Copyright 2018 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// Samples for the Container Analysis golang libnraries: https://cloud.google.com/container-registry/docs/container-analysis
+package main
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	containeranalysis "cloud.google.com/go/devtools/containeranalysis/apiv1beta1"
+	pubsub "cloud.google.com/go/pubsub"
+	"golang.org/x/net/context"
+	"google.golang.org/api/iterator"
+	grafeaspb "google.golang.org/genproto/googleapis/devtools/containeranalysis/v1beta1/grafeas"
+	"google.golang.org/genproto/googleapis/devtools/containeranalysis/v1beta1/vulnerability"
+)
+
+// [START create_note]
+
+//createNote creates and returns a new vulnerability Note.
+func createNote(ctx context.Context, client *containeranalysis.GrafeasV1Beta1Client, noteID, projectID string) (*grafeaspb.Note, error) {
+	parent := "projects/" + projectID
+
+	vul := vulnerability.Vulnerability{}
+	// Details about the your vulnerability can be added here.
+	// For Example: vul.Severity = vulnerability.Severity_CRITICAL
+	noteVul := grafeaspb.Note_Vulnerability{Vulnerability: &vul}
+	note := grafeaspb.Note{Type: &noteVul}
+	req := &grafeaspb.CreateNoteRequest{Parent: parent, NoteId: noteID, Note: &note}
+	return client.CreateNote(ctx, req)
+}
+
+// [END create_note]
+
+// [START create_occurrence]
+
+//createsOccurrence creates and returns a new Occurrence of a previously created vulnerability Note.
+func createOccurrence(ctx context.Context, client *containeranalysis.GrafeasV1Beta1Client, imageUrl, parentNoteID, projectID string) (*grafeaspb.Occurrence, error) {
+	noteName := "projects/" + projectID + "/notes/" + parentNoteID
+	projectName := "projects/" + projectID
+
+	details := vulnerability.Details{}
+	// Details about the vulnerability instance can be added here.
+	vulDetails := grafeaspb.Occurrence_Vulnerability{Vulnerability: &details}
+	// Attach the occurrence to the associated image uri.
+	resource := grafeaspb.Resource{Uri: imageUrl}
+	occurrence := grafeaspb.Occurrence{NoteName: noteName, Resource: &resource, Details: &vulDetails}
+	req := &grafeaspb.CreateOccurrenceRequest{Parent: projectName, Occurrence: &occurrence}
+	return client.CreateOccurrence(ctx, req)
+}
+
+// [END create_occurrence]
+
+// [START update_note]
+
+// updateNote pushes an update to a Note that already exists on the server.
+func updateNote(ctx context.Context, client *containeranalysis.GrafeasV1Beta1Client, updated *grafeaspb.Note, noteID, projectID string) error {
+	noteName := "projects/" + projectID + "/notes/" + noteID
+
+	req := &grafeaspb.UpdateNoteRequest{Name: noteName, Note: updated}
+	_, err := client.UpdateNote(ctx, req)
```

Comment:
Should return the updated note in this function as well, just like you do with returning the created note.